common/cloudant_base_test.go

```diff
@@ -167,4 +167,18 @@ var _ = Describe(`Cloudant custom base service UT`, func() {
 		Expect(authenticator).ToNot(BeNil())
 		Expect(authenticator.AuthenticationType()).To(Equal(core.AUTHTYPE_IAM))
 	})
+
+	It("Validates URL trailing slash is stripped", func() {
+		cloudant, err := NewBaseService(&core.ServiceOptions{
+			URL:           "https://cloudant.example/",
+			Authenticator: &core.NoAuthAuthenticator{},
+		})
+		Expect(cloudant).ToNot(BeNil())
+		Expect(err).To(BeNil())
+
+		builder := core.NewRequestBuilder(core.GET)
+		_, err = builder.ResolveRequestURL(cloudant.Options.URL, "/db", map[string]string{})
+		Expect(err).To(BeNil())
+		Expect(builder.URL.String()).To(Equal("https://cloudant.example/db"))
```

Comment:
It would be good to add an another case for `GetServerInformation` case:
```go
		_, err = builder.ResolveRequestURL(cloudant.Service.Options.URL, `/`, nil)
```

Reply:
No, this wouldn't work, because `go-sdk-core` has [unusual logic](https://github.com/IBM/go-sdk-core/blob/main/v5/core/request_builder.go#L167-L181) to deal with slashes in url - if `serviceURL` has trailing slash it trims prefix slash from the `path` and if `serviceURL` doesn't have trailing slash, it adds leading slash to the `path` when it's missing. So your example will end-up _with_ trailing slash in `builder.URL`, because it'll be first removed from `path`, but then kept as suffix on `urlString`.

Basically while all other core sdks are trimming trailing slash from `serviceURL`, `go-sdk-core` is doing it differently and de-duplicating slashes while constructing URL instead, that's why using "/db" here is more visible, it is to confirm de-duplication, that we will not end up with "https://cloudant.example//db" when have both trailing slash on service url and leading slash on path.

Reply:
Thanks for the explanation. Now I get this.